safehttp/mux.go

```diff
@@ -167,27 +195,26 @@ func (s *ServeMuxConfig) Mux() *ServeMux {
 		panic("Use NewServeMuxConfig instead of creating ServeMuxConfig using a composite literal.")
 	}
 	// pattern -> method -> handlerConfig
-	handlers := map[string]map[string]handlerConfig{}
+	handlers := make(map[string]map[string]*handlerConfig)
 	for _, hr := range s.handlers {
 		if handlers[hr.pattern] == nil {
-			handlers[hr.pattern] = map[string]handlerConfig{}
-		}
-		if _, ok := handlers[hr.pattern][hr.method]; ok {
-			// TODO: this should be done in ServeMuxConfig.Handle, not here.
-			panic(fmt.Sprintf("double registration of (pattern = %q, method = %q)", hr.pattern, hr.method))
+			handlers[hr.pattern] = make(map[string]*handlerConfig)
 		}
-		handlers[hr.pattern][hr.method] =
-			handlerConfig{
-				Dispatcher:   s.dispatcher,
-				Handler:      hr.handler,
-				Interceptors: configureInterceptors(s.interceptors, hr.cfgs),
-			}
+		handlers[hr.pattern][hr.method] = s.handlerRegistrationToHandlerConfig(&hr)
```

Comment:
Isn't `&hr` a bug here? Aren't you taking a pointer to the same local variable?

Reply:
Good question, I was hoping for a test to catch that, but I will take another look.

Reply:
In this case it isn't because you are not storing `&hr` anywhere, but it is definitely error-prone.